webpack/components/NewTemplateSync/components/SyncSettingFields.js

```diff
@@ -58,14 +42,15 @@ const SyncSettingsFields = ({
   return (
     <React.Fragment>
       {settingsAry
-        .map(setting => addValidationToSetting(setting, validationData))
+        .map(addRequiredToSetting)
```

Comment:
The way it works is a bit odd, the `SyncSettingsFields` receives `importSettings`, `exportSettings` and `syncType`.
Based on the `syncType` it will use `importSettings` or `exportSettings`.

IMO it should be done at a higher level so this component would only receive the final `settingsAry`.
Same for the manipulations done on the `settingsAry`, it can all be done in a **memorized selector**.

Please consider refactoring, maybe it should be done in a different PR.